lib/data_indexer.rb

```diff
@@ -0,0 +1,112 @@
+require 'csv'
+class DataIndexer
+
+  # Index mods and annotations data from a list of manifest urls or each url
+  # Params:  
+  # +collection+:: Name of collection the manifest(s) belong to
+  # +csv_file+:: string containing the path to the csv file. 
+  #    File to have one url per line and no header
+  # +manifest_url+:: url to the mnaifest file
+  # Usage: 
+  #   DataIndexer.new('collection_name', 'file_path').run 
+  #      to index csv file
+  #   DataIndexer.new('collection_name', nil, 'url').run
+  #      to index one manifest at url
+  def initialize(collection = nil, csv_file = nil, manifest_url = nil)
+    @collection = collection
+    @csv_file = csv_file
+    @url = manifest_url
+    @manifest = nil
+    @title = nil
+    @doc = SolrDocument.new
+    @solr = Blacklight.default_index.connection
+  end
+
+  # Index and commit mods and annotations data either 
+  # from a list of manifest urls or each url
+  # depending on the options 
+  def run
+    if !@csv_file.blank? && File.exist?(@csv_file)
```

Comment:
Perhaps more of a stylistic thing but `@csv_file.present?` reads a little better than `!@csv_file.blank?` to me.  Totally up to you though.